clar2wasm/src/standard/standard.wat

```diff
@@ -2522,6 +2536,293 @@
         (local.get $hi)
     )
 
+
+    (func $stdlib.sha512-buf (param $offset i32) (param $length i32) (param $offset-result i32)(result i32 i32)
+
+        ;; Length of the data after adding padding and length to it
+        (local $length_after_padding i32)
+
+        (local $message_length_64 i64)
+        
+        ;; Index to track all blocks
+        (local $outer-index i32) 
+        
+        ;; Temporary block data (a word of 8 bytes) in current block iteration
+        (local $block-iteration-temp i64)
+
+        ;; Local index to a block, to perform iterations on bytes
+        (local $inner-index i32)
+
+        ;; Temporary block data, in an iteration
+        (local $temp-block-data i32)
+
+        ;; Variables used in hash rounds
+        (local $a i64) (local $b i64) (local $c i64) (local $d i64)
+        (local $e i64) (local $f i64) (local $g i64) (local $h i64)
+        (local $temp1 i64) (local $temp2 i64)
+
+        
+
+        ;; Copying initial values (64 bytes) for SHA-512 from 648 index
+        (memory.copy (global.get $stack-pointer) (i32.const 648) (i32.const 64))
+
+        ;; Copying the data from the offset to isolated environment (i.e. target-index = $stack-pointer+(initial-values+(80 rounds * 8)))
+        (memory.copy (i32.add (global.get $stack-pointer) (i32.const 704)) (local.get $offset) (local.get $length))
+
+        (local.set $length_after_padding ;; total size of data with expansion divisible by 128
+            (i32.add
+                (i32.or
+                    ;; len + 16 bytes for the size
+                    (i32.add (local.get $length) (i32.const 16))
+                        (i32.const 0x7f)
+                    )
+                    (i32.const 1)
+                )
+        )
+
+        
+        ;; Add "1" at the end of the data
+        (i32.store offset=704
+            (i32.add (global.get $stack-pointer) (local.get $length))
+            (i32.const 0x80)
+        )
+
+        ;; Fill the remaining part before the size with 0s
+        (memory.fill
+                (i32.add (i32.add (global.get $stack-pointer) (local.get $length)) (i32.const 705))
+                (i32.const 0)
+
+                ;; Leave the last 8 bytes for the length
+                ;; Not handling 16 bytes length here
+                (i32.sub (i32.sub (local.get $length_after_padding) (local.get $length)) (i32.const 8))
+        )
+
+        ;; Add the length, as a 64bits big-endian integer
+        (local.set $message_length_64 (i64.extend_i32_u (i32.shl (local.get $length) (i32.const 3))))
+
+
+        ;; Although length is being handled as i64 (8 bytes)
+        ;; This is the location where reversed length is going to be stored
```

Comment:
Although? What does this comment mean?

Reply:
I was editing that comment, somehow I left it there. Fixed